File: client/src/app/shared/models/approvalTimeLimit.ts
import { IApprovalAuthority } from "./approvalAuthority";

export interface IApprovalTimeLimit {
    id: number;
    approvalAuthorityId: number;
    timeLimit: number;
    remarks: string;
    status: string;
    setOn: Date;
    approvalAuthority:IApprovalAuthority;
}


export class ApprovalTimeLimit implements IApprovalTimeLimit {
    id: number = 0;
    approvalAuthorityId: number=null;
    timeLimit: number=null;
    remarks: string;
    status: string=null;
    setOn: Date;
    approvalAuthority:IApprovalAuthority;
}

File: Core/Entities/ApprovalTimeLimit.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities
{
    public class ApprovalTimeLimit:BaseEntity
    {
        public int ApprovalAuthorityId { get; set; }
        [ForeignKey("ApprovalAuthorityId")]
        public ApprovalAuthority ApprovalAuthority { get; set; }
        public int TimeLimit { get; set; }
        public string Remarks { get; set; }
        public string Status { get; set; }    
    }
}